src/compiler/program.ts

```diff
@@ -1052,7 +1056,7 @@ export function loadWithModeAwareCache<Entry, SourceFile, ResolutionCache, Resol
     const loader = createLoader(containingFile, redirectedReference, options, host, resolutionCache);
     for (const entry of entries) {
         const name = loader.nameAndMode.getName(entry);
-        const mode = loader.nameAndMode.getMode(entry, containingSourceFile);
+        const mode = loader.nameAndMode.getMode(entry, containingSourceFile, options);

```

Comment:
I think you want to use `redirectedReference.options || options` since for imports in file from referenced project, we want to use the options of referenced project and not referencing project

Reply:
I think the expected behavior is debatable. Suppose you have a bundled frontend app that references a shared library designed to work in Node.js as well as bundlers and other runtimes. The referenced project settings would probably use module settings for Node.js, since those tend to be the strictest. But when the bundled frontend app imports that project, the actual module system in play is only the bundler—it doesn’t matter what settings the referenced library was checked against. In this scenario, it makes more sense to use the options from the primary project, not the one being imported.

But you can also imagine the opposite. Suppose you have a Node.js app that references a shared library built into a single file with a bundler. In that case, the primary project (the Node.js app) would be using very strict, Node.js-specific module settings, while the referenced library might have been built with `--moduleResolution bundler` and maybe even some `customConditions`. And whatever imports are present in the source files / declaration files of the bundled library have already been baked into the output that the Node.js app will see at runtime. In this scenario, it makes more sense to use the options from the referenced project.

Both of these sound plausible, and we don’t have configuration sophisticated enough to distinguish between them. I think the former sounds more likely, which is why I chose to use the options of the primary project being compiled. However, I’m ok with switching if there’s a strong precedent to do everything with respect to the referenced project options.

Reply:
When resolving imports from references projects it’s going to be .d.ts file built from referenced project but resolutions happen at original ts location with the referenced project setting…in fact that’s how it works today as otherwise the resolutions may not work 

for editor scenarios anyways we include ts file from referenced project and all the resolutions in that project follow the options of referenced project 

when I get to desk tomorrow I can link the issue where we started doing this

Reply:
Also note that none of the imports are emitted so emit thing doesn’t matter